lightning/src/events/bump_transaction.rs

```diff
@@ -231,3 +267,486 @@ pub enum BumpTransactionEvent {
 		tx_lock_time: PackedLockTime,
 	},
 }
+
+/// An input that must be included in a transaction when performing coin selection through
+/// [`CoinSelectionSource::select_confirmed_utxos`].
+pub struct Input {
+	/// The unique identifier of the input.
+	pub outpoint: OutPoint,
+	/// The upper-bound weight consumed by the input's full witness required to satisfy its
+	/// corresponding output's script.
+	pub witness_weight: u64,
+}
+
+/// An unspent transaction output that is available to spend resulting from a successful
+/// [`CoinSelection`] attempt.
+#[derive(Clone, Debug)]
+pub struct Utxo {
+	/// The unique identifier of the output.
+	pub outpoint: OutPoint,
+	/// The output to spend.
+	pub output: TxOut,
+	/// The upper-bound weight consumed by the corresponding input's full witness required to
+	/// satisfy the output's script.
+	pub witness_weight: u64,
+}
+
+/// The result of a successful coin selection attempt for a transaction requiring additional UTXOs
+/// to cover its fees.
+pub struct CoinSelection {
+	/// The set of UTXOs (with at least 1 confirmation) to spend and use within a transaction
+	/// requiring additional fees.
+	confirmed_utxos: Vec<Utxo>,
```

Comment:
See my feedback above, unconfirmed utxos could be use in case of lacking UTXOs to fee-bump, though maybe we can add the usage of unconfirmed utxos feature-gated behind its own `select_confirmed_utxos_internal` boolean in future PRs ?